metrics/module.go

```diff
@@ -0,0 +1,131 @@
+package metrics
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/safing/portbase/modules"
+)
+
+var (
+	module *modules.Module
+
+	registry     []Metric
+	registryLock sync.RWMutex
+
+	firstMetricRegistered bool
+	metricNamespace       string
+	globalLabels          = make(map[string]string)
+
+	pushURL string
+
+	// ErrAlreadyStarted is returned when an operation is only valid before the
+	// first metric is registered, and is called after.
+	ErrAlreadyStarted = errors.New("can only be changed before first metric is registered")
+
+	// ErrAlreadyRegistered is returned when a metric with the same ID is
+	// registered again.
+	ErrAlreadyRegistered = errors.New("metric already registered")
+
+	// ErrAlreadySet is returned when a value is already set and cannot be changed.
+	ErrAlreadySet = errors.New("already set")
+)
+
+func init() {
+	flag.StringVar(&pushURL, "push-metrics", "", "URL to push prometheus metrics to")
+
+	module = modules.Register("metrics", prep, start, stop, "database", "api")
+}
```

Comment:
Should we allow the user to configure global labels per commandline flags. Usage scenario would be to get metrics from various portmaster installations pushed to my central push-gateway. I could add a global label to identify the different hosts.

Do we want to be able to configure the metrics push-gateway from the UI as well?